packages/forms/src/Components/Builder.php

```diff
@@ -663,6 +669,8 @@ public function getEditAction(): Action
             ->label(__('filament-forms::components.builder.actions.edit.label'))
             ->modalHeading(__('filament-forms::components.builder.actions.edit.modal.heading'))
             ->modalSubmitActionLabel(__('filament-forms::components.builder.actions.edit.modal.actions.save.label'))
+            ->closeModalByClickingAway(fn (Builder $component): bool => ! $component->isModalPersistent())
+            ->closeModalByEscaping(fn (Builder $component): bool => ! $component->isModalPersistent())
```

Comment:
I introduced single property for both click outside and ESC dialog close. I don't see use case when someone want one, but don't want another, and I wanted to keep it simple.